src/app/catalogue-search/catalogue-search-advanced/catalogue-search-advanced-form.component.ts

```diff
@@ -0,0 +1,143 @@
+SPDX-License-Identifier: Apache-2.0
+*/
+import { Component, OnInit } from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+import {
+  Classifier,
+  ClassifierIndexResponse,
+  MdmTreeItem
+} from '@maurodatamapper/mdm-resources';
+import { MdmResourcesService } from '@mdm/modules/resources';
+
+@Component({
+  selector: 'mdm-catalogue-search-advanced-form',
+  templateUrl: './catalogue-search-advanced-form.component.html',
+  styleUrls: ['./catalogue-search-advanced-form.component.scss']
+})
+export class CatalogueSearchAdvancedFormComponent implements OnInit {
+  advancedSearch: boolean;
+  formGroup: FormGroup;
+  classifications: Classifier[];
+
+  get context() {
+    return this.formGroup.get('context');
+  }
+
+  set contextValue(value: MdmTreeItem[]) {
+    this.context.setValue(value);
+  }
+
+  get classifiers() {
+    return this.formGroup.get('classifiers');
+  }
+
+  get classifierNames() {
+    if (!this.classifiers.value) {
+      return;
+    }
+
+    const classifierNames = [];
+    this.classifiers.value.forEach((classifier) => {
+      classifierNames.push(classifier.label);
+    });
+    return classifierNames;
+  }
+
+  get domainTypes() {
+    return this.formGroup.get('domainTypes');
+  }
+
+  get labelOnly() {
+    return this.formGroup.get('labelOnly');
+  }
+
+  get exactMatch() {
+    return this.formGroup.get('exactMatch');
+  }
+
+  get lastUpdatedAfter() {
+    return this.formGroup.get('lastUpdatedAfter');
+  }
+
+  get lastUpdatedBefore() {
+    return this.formGroup.get('lastUpdatedBefore');
+  }
+
+  get createdAfter() {
+    return this.formGroup.get('createdAfter');
+  }
+
+  get createdBefore() {
+    return this.formGroup.get('createdBefore');
+  }
+
+  ngOnInit(): void {
+    this.formGroup = new FormGroup({
+      context: new FormControl(null),
+      domainTypes: new FormControl(''),
+      labelOnly: new FormControl(true),
+      exactMatch: new FormControl(false),
+      classifiers: new FormControl(''),
+      createdAfter: new FormControl(null),
+      createdBefore: new FormControl(null),
+      lastUpdatedAfter: new FormControl(null),
+      lastUpdatedBefore: new FormControl(null)
+    });
+
+    this.resources.classifier
+      .list({ all: true })
+      .subscribe((result: ClassifierIndexResponse) => {
+        this.classifications = result.body.items;
+      });
+  }
+
+  constructor(private resources: MdmResourcesService) {}
+
+  toggleAdvancedSearch() {
+    this.advancedSearch = !this.advancedSearch;
+  }
+
+  formatDate(date: Date) {
+    if (!date) {
+      return;
+    }
+
+    const yyyy: String = date.getFullYear().toString();
+    const mm: String = date.getMonth().toString().padStart(2, '0');
+    const dd: String = date.getDate().toString().padStart(2, '0');
+
+    return `${yyyy}-${mm}-${dd}`;
+  }
+
+  onDateClear(control: string) {
+    this.formGroup.get(control).setValue(null);
+  }
+
+  reset() {
+    this.formGroup.controls.context.setValue(null);
```

Comment:
I've noticed that clicking the "Reset" button doesn't actually clear the context selection tree control. Because that is a custom control, perhaps you need to manually reset the inputs for it.

Reply:
Did not fix for now, using the MDMTreeSearchComponent as a view child binds the components together and want to avoid modifying MDMTreeSearchComponent if possible. Could also use the broadcast service but that seems out of scope for a small change